### PR TITLE
[doc] Small elaboration for Docker image usage

### PR DESCRIPTION
**Pull request by 130s**
Not sure how effective this change is but trying to address confusion such as #212.

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -190,6 +190,20 @@ Note that some of these currently tied only to a single option, but we still lea
...
+Use custom Docker images
+------------------------
+
+As you see in the `optional variables section <./index.rst#optional-environment-variables>`_, there are a few different ways to specify `Docker` image if you like. Here are some more detail:
+
+Pulling Docker image from an online hub
++++++++++++++++++++++++++++++++++++++++
+
+You can use `DOCKER_IMAGE` variable for this.
+
```
Comment (ipa-mdl) [inline doc/index.rst]: Perhaps we should add a not on the requirements that such an image have to meet:
* wstool
* rosdep; including repository set-up und python-pip
* catkin-tools

File: industrial_ci/src/tests/source_tests.sh [hunk 2]
```diff
@@ -39,8 +39,13 @@ ROSWS=wstool
 ici_time_end  # init_ici_environment
 
 function catkin {
+  PATH_CATKIN=/usr/bin/catkin
```
Comment (ipa-mdl) [inline industrial_ci/src/tests/source_tests.sh]: Variable should be kept local. Local variables should be lowercase.

File: industrial_ci/src/tests/source_tests.sh [hunk 3]
```diff
@@ -39,8 +39,13 @@ ROSWS=wstool
...
+  PKG_CATKIN=python-catkin-tools
   local cmd=$1
   shift
+  if [ ! -f ${PATH_CATKIN} ]; then
+  	echo "[ERR] ${PATH_CATKIN} not available. Make sure ${PKG_CATKIN} is installed. See also https://github.com/ros-industrial/industrial_ci/issues/216";
```
Comment (ipa-mdl) [inline industrial_ci/src/tests/source_tests.sh]: The error function should be used instead. I will take care of the colors and the exit code.

**Comment by ipa-mdl** [comment]
I took the chance to improve you're warning code.
The new version works with other paths to `catkiin`.

File: industrial_ci/src/tests/source_tests.sh (hunk 3 above)
Comment (130s) [reply industrial_ci/src/tests/source_tests.sh]: Yeah, I just couldn't find `err` was used in the same .sh file so I tried not to use it.

**Comment by 130s** [comment]
@ipa-mdl thanks, your change looks good.
How about the earlier commits I made?

File: doc/index.rst (hunk 1 above)
Comment (130s) [reply doc/index.rst]: > rosdep; including repository set-up und python-pip

What is repository set-up? `rosdep init && rosdep update`?

File: doc/index.rst [hunk 4]
```diff
@@ -190,6 +190,25 @@ Note that some of these currently tied only to a single option, but we still lea
...
+As you see in the `optional variables section <./index.rst#optional-environment-variables>`_, there are a few different ways to specify `Docker` image if you like. Here are some more detail:
+
+Pulling Docker image from an online hub
++++++++++++++++++++++++++++++++++++++++
+
+You can pull any `Docker` image by specifying in `DOCKER_IMAGE` variable, as long as the following requirement is ment:
+
+* `python-catkin-tools`, `python-pip`, `python-wstool`
+* `python-rosdep` including rosdep repository set-up
```
Comment (ipa-mdl) [inline doc/index.rst]: rosdep repository is not required, but ROS repo set-up is.

Comment (130s) [reply doc/index.rst]: Updated.

**Comment by 130s** [comment]
> Something went wrong with the last to commit, 0b8edca is now duplicated.

I don't see it duplicated (you may have already fixed it?) but now I don't see an issue?

**Comment by ipa-mdl** [comment]
https://github.com/ros-industrial/industrial_ci/pull/215/commits/7bc53558b3d2b637e0d54d031aa98a4b72de88ff should not be part of this PR, it's the same patch..

**Comment by 130s** [comment]
@ipa-mdl sorry. I rebased and now it should look good.

**Comment by ipa-mdl** [comment]
I think some commits could be squashed, but it is not required.
Feel free to shape it or merge right-away :)

**Comment by 130s** [comment]
Some are squashed and CI passed. Thanks @ipa-mdl

**Pull request merged**